src/js/bun/ffi.ts

```diff
@@ -117,20 +117,17 @@ class JSCallback {
 
 class CString extends String {
   constructor(ptr, byteOffset?, byteLength?) {
+    byteOffset ||= 0;
     super(
       ptr
         ? typeof byteLength === "number" && Number.isSafeInteger(byteLength)
-          ? BunCString(ptr, byteOffset || 0, byteLength)
-          : BunCString(ptr, byteOffset || 0)
+          ? BunCString(ptr, byteOffset, byteLength)
+          : BunCString(ptr, byteOffset)
         : "",
     );
+    this.byteLength = byteLength || Buffer.byteLength(this.toString(), 'utf-8');
```

Comment:
This adds overhead to every usage of CString for something rarely needed. Instead, can this be a getter and we cache the value with a #private field?

Reply:
I was hoping that I was missing a more… performant… approach, but I've implemented your suggestion.

Let me know what you think!